### DISCUSSION TITLE
Generating Mesh files in SPECFEM Cartesian

### DISCUSSION THREAD

**Discussion by Rockhound88** [SPECFEM/specfem3d#1617] [SPECFEM3D General]
Hello,

I'm not new to SPECFEM, but am to SPECFEM_Cartesian. I have started to construct my own model to run, however I seem to be running into an error that is not displaying any hints that I could use to directly address the problem (at least to my knowledge). 
When I run the ./bin/xmeshfem3D command, my run abruptly stops and in my output_meshfem3D.txt file it stops by displaying:

"saving mesh files"

However, when I run the socal1D example I see that the file is supposed to continue on by entering another section:

"saving mesh files"

 **************************
 Checking mesh quality
 **************************

-------------
I know the issue has to do with the proc00000X_mesh.INP files in the DATABASES_MPI directory (namely that there are none). However, I don't know what the issue is (I apologize if it is obvious). Is this simply a case where I gave my dimension partitions incorrectly or something similar?

**Comment by aakash10gupta** [comment]
Hi @Rockhound88, if you can share your input data files, I can try having a look.

**Comment by Rockhound88** [comment]
Hi @aakash10gupta , thanks for the response. Are these the input files you mean? 
[interfaces.txt](https://github.com/SPECFEM/specfem3d/files/11697121/interfaces.txt)
[Mesh_Par_file.txt](https://github.com/SPECFEM/specfem3d/files/11697122/Mesh_Par_file.txt)
[Par_file.txt](https://github.com/SPECFEM/specfem3d/files/11697123/Par_file.txt)

From my understanding, the partitions of the different regions should be correct, I am simply trying to expand the example model into 9 different levels as opposed to 3 just to understand how the system works. I am assuming I am missing something pretty basic.

**Comment by aakash10gupta** [comment]
Hi @Rockhound88, your Mesh_Par_file needed a few changes based on the inputs in your interfaces file. The updated Mesh_Par_file is attached below. You will also need interface files corresponding to each interface that you defined in your interfaces file. I am attaching the ones I created to run your test case.

[interface1.txt](https://github.com/SPECFEM/specfem3d/files/11702574/interface1.txt)
[interface2.txt](https://github.com/SPECFEM/specfem3d/files/11702576/interface2.txt)
[interface3.txt](https://github.com/SPECFEM/specfem3d/files/11702577/interface3.txt)
[interface4.txt](https://github.com/SPECFEM/specfem3d/files/11702578/interface4.txt)
[interface5.txt](https://github.com/SPECFEM/specfem3d/files/11702579/interface5.txt)
[interface6.txt](https://github.com/SPECFEM/specfem3d/files/11702580/interface6.txt)
[interface7.txt](https://github.com/SPECFEM/specfem3d/files/11702581/interface7.txt)
[interface8.txt](https://github.com/SPECFEM/specfem3d/files/11702582/interface8.txt)
[interface9.txt](https://github.com/SPECFEM/specfem3d/files/11702583/interface9.txt)
[Mesh_Par_file.txt](https://github.com/SPECFEM/specfem3d/files/11702666/Mesh_Par_file.txt)

Note that you will have to modify the file extensions as needed.

I hope identifying the changes in the attached files would explain the several reasons why the mesh generation process was failing. Let me know if anything is not clear.
